besu/src/main/java/com/blockchaintp/besu/daml/ValueCacheSingleton.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 Blockchain Technology Partners
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.blockchaintp.besu.daml;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.google.protobuf.ByteString;
+
+public class ValueCacheSingleton {
+  private static final String VALUE_CACHE_SIZE_ENV = "VALUE_CACHE_SIZE";
+  private static final String DEFAULT_CACHE_SIZE = "100";
+  private static ValueCacheSingleton instance;
+  private Cache<ByteString, ByteString> cache;
+
+  public Cache<ByteString, ByteString> getCache() {
+    return cache;
+  }
+
+  private ValueCacheSingleton() {
+    var env = System.getenv();
+    long cacheSz = Long.parseLong(env.getOrDefault(VALUE_CACHE_SIZE_ENV, DEFAULT_CACHE_SIZE));
```

Comment:
I would prefer cacheSize, or as the context is caching just 'size'. Abbreviation schemes tend to be quite subjective. I read 'Sz' as string due to years of https://en.wikipedia.org/wiki/Hungarian_notation (now thankfully very out of fashion)